site/learn/expert-advice/authentication/webauthn-explained.md

```diff
@@ -0,0 +1,478 @@
+
+## User Presence vs User Verification
+
+Authenticators can be also used as a second factor of authentication, when paired with a username and password or other factor. An authentication ceremony gives you assurances about the user having access to the authenticator. If you are using an authenticator in this way, you want to distinguish between user presence versus user verification. 
+
+User presence is when someone is there behind the authenticator who gave permission, but you don't know who. All it proves is that you have access to the authenticator. An example of this is when you press a button on a Yubikey.
+
+User verification, on the other hand, is when the user provides proof that they are who they say they are. The authenticator could require a pin, known only to the user. It could be a Face ID or Touch ID challenge. In any case, verification allows you to tie a specific user to the authentication ceremony.
+
+Of the two types, user verification is more prevalent with the platform authenticators than cross platform authenticators. And user presence is more useful as a second factor. When not used for MFA, you will typically want user verification.
+
+The initiating server specifies whether to use user verification. The three choices for user verification are preferred, required, and discouraged:
+
+* Preferred means that user verification is preferred, but not required. 
+* Required means that user verification is required and the authentication will fail if it cannot be performed. 
+* Discouraged means that user verification is not preferred and should not be used if possible. 
+
+## Resident Keys
+
+Another WebAuthn feature is resident keys. With these, the user to website mapping is stored, not on the server side, but on the authenticator.
+
+This is available for cross-platform authenticators. With this feature, the server does not have to have prior knowledge of the user before they are authenticated. They can register and authenticate at the same time. This is less common in the B2C space.
+
+## Implementation Details
+
+What do you need to consider when implementing WebAuthn authentication? In some ways, the actual ceremonies are easy; they are specified in a fair bit of detail in the W3C standard. One FusionAuth developer said "the hardest part of the WebAuthn portion was translating from byte arrays to base64 and back".
+
+But there's more, such as:
+
+- Storing public key data in a safe and secure way
+- Streamlining workflows, such as knowing when to prompt the user for WebAuthn
+- Prompting users to register their WebAuthn capable device
+- Allowing users to use WebAuthn during the app sign in process
+- Considering the account recovery process
+
+There are libraries which can help with the server side management and WebAuthn prompts. These libraries can help with tasks like storing data safely and securely, streamlining workflows, and prompting users to register.
```

Comment:
Most of the libraries I saw were very focused on generating, translating, and validating the WebAuthn request options and response objects. I think it's up to the RP to figure out all the other stuff because it's going to vary so much between companies.